File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.java
import java.util.HashMap;
import java.util.Map;

class Solution {
	public String[] solution(String[] players, String[] callings) {	
		
		// HashMap Setting
		Map<String, Integer> indexMap = new HashMap<>();
		for(int i = 0; i < players.length; i++) {
			indexMap.put(players[i], i);
		}
		
		for(String calledPlayer : callings) {
			int index = indexMap.get(calledPlayer);
			
			// players 배열 갱신
			String front = players[index - 1];
			
			players[index - 1] = calledPlayer;
			players[index] = front;
			
			//indexMap 갱신
			indexMap.put(calledPlayer, index - 1);
			indexMap.put(front, index);
		}
		return players;
	}
}

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {

    let playersMap = new Map();

    // players를 Map으로 재생성
    for(let i = 0; i < players.length; i++) {
        playersMap.set(players[i], i);        
    }

    for(let str of callings) {
        let index = playersMap.get(str);

        // players 배열 갱신
        let front = players[index - 1] 
        let back = players[index]

        players[index - 1] = back;
        players[index] = front;

        // playersMap 갱신
        playersMap.set(back, index - 1);
        playersMap.set(front, index);
    }
    return players;
}